### PR TITLE
Fix en doc for rnn.py. test=document_fix

### PR DESCRIPTION
**Pull request by smallv0221**
### PR types

Others 
### PR changes

Docs 
### Describe
Fix equation and example errors and add example output for en doc in rnn.py.

### Preview

http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/en/api/paddle/nn/GRUCell_en.html
image

http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/en/api/paddle/nn/GRU_en.html
image

http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/en/api/paddle/nn/LSTM_en.html
image

http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/en/api/paddle/nn/LSTMCell_en.html
image

http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/en/api/paddle/nn/RNN_en.html
image

http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/en/api/paddle/nn/BiRNN_en.html
image

http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/en/api/paddle/nn/SimpleRNNCell_en.html
image

### CODE REVIEW

File: python/paddle/nn/layer/rnn.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def get_initial_states(self,
                 states. For `batch_ref`'s shape d, `d[batch_dim_idx]` is 
                 treated as batch size.
             shape (list|tuple, optional): A (possibly nested structure of) shape[s], 
-                where a shape is a list/tuple of integer). `-1` (for batch size) 
+                where a shape is a list/tuple of integer. `-1` (for batch size)
```
Comment (TCChenlong) [inline python/paddle/nn/layer/rnn.py:164]: 157、177前面需要加个空行 不然会有显示bug

File: python/paddle/nn/layer/rnn.py [hunk 2]
```diff
@@ -154,13 +154,14 @@ def get_initial_states(self,
         r"""
         Generate initialized states according to provided shape, data type and
         value.
+
```
Comment (TCChenlong) [inline python/paddle/nn/layer/rnn.py:157]: 参数统一为 `Parameters`，变量统一为`Variables`

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by iclementine — APPROVED** [review]
LGTM!

**Pull request merged**